libs/go/ztsroletoken/role-token.go

```diff
@@ -0,0 +1,177 @@
+package roletoken
+
+import (
+	"sync"
+	"time"
+
+	"crypto/tls"
+	"errors"
+	"io/ioutil"
+	"net/http"
+
+	"crypto/x509"
+	"fmt"
+
+	"github.com/yahoo/athenz/clients/go/zts"
+	"github.com/yahoo/athenz/libs/go/zmssvctoken"
+)
+
+const (
+	defaultPrincipalAuthHeader = "Athenz-Principal-Auth"
```

Comment:
Having this as a const, will it prohibit the use in a different environment with a different header name?